src/Compilers/CSharp/Portable/Syntax/SyntaxKindFacts.cs

```diff
@@ -1046,12 +1048,25 @@ public static SyntaxKind GetPreprocessorKeywordKind(string text)
             }
         }
 
+        private static ReadOnlyCollection<SyntaxKind> _lazyContextualKeywordKindsArray;
```

Comment:
As i mentioned:

> I understood the value in you change to change the constant so that you could add contextual keywords withou thaving to update this. 

What i'm pushing back on is teh change in overal logic to now lazily check for and cache some result list.  :)

> There is no comment (at least) in the keyword enum that tells me that this function has also to be updated. 

You could always just add that comment instead.  :)